config/routes.rb

```diff
@@ -1,6 +1,12 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get '/', to: 'users#user_data'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get 'users/:user_id/posts', to: 'users#show_posts', as: 'user_posts'
+  get 'users/:user_id/posts/:post_id', to: 'users#show_post', as: 'user_post'
```

Comment:
- [ ] Kindly update your routes with the corresponding post and index methods when you do implement as I stated above 

### Expected from requirement:
![image](https://github.com/Samuelwanza/sema/assets/50990925/c4905cc3-d378-42d7-82db-02066e36bda3)